wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,419 @@
+            view_perm = channel.overwrites_for(ctx.guild.default_role).view_channel
+
+            # skip private channels and read-only channels
+            if view_perm is False or send_perm is False:
+                continue
+            else:
+                await channel.purge(limit=100, check=is_banned_user, after=date, bulk=True)
+
+    @commands.command()
+    async def unban(self, ctx, user_id: int):
+        """Unbans a user"""
+
+        logger.info(f"[Ban unban()] unban command detected from {ctx.author} with args=[ {user_id} ]")
+        if user_id not in self.ban_list:
+            logger.info(f"Provided id: {user_id}, does not belong to a banned member.")
+            e_obj = await em(ctx, title="Error",
+                             content=[("Problem",
+                                       f"`{user_id}` is either not a valid Discord ID **OR** is not a banned user.")],
+                             colour=self.error_colour,
+                             footer="Command Error")
+            if e_obj:
+                await ctx.send(embed=e_obj)
+            return
+
+        # "unban"
+        self.ban_list.remove(user_id)
+
+        name = await BanRecords.unban_by_id(user_id)
+        if not name:
+            logger.info(f"[Ban unban()] No user with id: {user_id} found.")
+            await self.mod_channel.send(f"*No user with id: **{user_id}** found.*")
+
+        logger.info(f"[Ban unban()] User: {name} with id: {user_id} was unbanned.")
+        e_obj = await em(ctx, title="Unban", description=f"**`{name}`** was unbanned.", colour=discord.Color.red())
+        if e_obj:
+            await self.mod_channel.send(embed=e_obj)
+
+    @unban.error
+    async def unban_error(self, ctx, error):
+        """Catches an error in unban when a non integer is passed in as an argument"""
+
+        logger.info("[Ban Unban_error] caught non integer ID passed into unban parameter. Handled accordingly")
+        if isinstance(error, commands.BadArgument):
+            e_obj = await em(ctx, title="Error",
+                             content=[("Problem", "Please enter a numerical Discord ID.")],
+                             colour=self.error_colour,
+                             footer="Command Error")
+            if e_obj:
+                await ctx.send(embed=e_obj)
+
+    @commands.command()
+    async def bans(self, ctx):
+        """Gets all banned users"""
+
+        logger.info(f"[Ban bans()] bans command detected from {ctx.author}")
+
+        bans = await BanRecords.get_all_active_bans()
+        logger.info(f"[Ban bans()] retrieved all banned users: {bans}")
+
+        emb = discord.Embed(title="Banned members", color=discord.Color.red())
+
+        names = ""
+        ids = ""
+        for ban in bans:
+            # name = ban[0]
+            # user_id = ban[1]
+            name = ban['username']
+            user_id = ban['user_id']
+            if len(names) + len(name) > 1024 or len(ids) + len(str(user_id)) > 1024:
```

Comment:
I think this would be easier to read like this `if len(names) + len(name) > 1024 or (len(ids) + len(str(user_id))) > 1024:`